src/Service/Services/MetadataProviders/ISqlMetadataProvider.cs

```diff
@@ -108,6 +108,15 @@ public interface ISqlMetadataProvider
         /// <returns>True if exists, otherwise false.</returns>
         bool TryGetEntityNameFromPath(string entityPathName, [NotNullWhen(true)] out string? entityName);
 
+        /// <summary>
+        /// Try to obtain the path of the Entity that has the provided entityName. If It
+        /// exists save in out param, and return true, otherwise return false.
+        /// </summary>
+        /// <param name="entityName">Name of the associated entity.</param>
+        /// <param name="entityPathName">Entity's path as seen in a request.</param>
+        /// <returns>True if exists, otherwise false.</returns>
+        bool TryGetEntityPathFromName(string entityName, [NotNullWhen(true)] out string? entityPathName);
```

Comment:
You could avoid having to implement this on the Cosmos provider by using a [default interface implementation](https://devblogs.microsoft.com/dotnet/default-implementations-in-interfaces/).

```suggestion
        bool TryGetEntityPathFromName(string entityName, [NotNullWhen(true)] out string? entityPathName) => throw new NotImplementedException();
```